# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Marking.cpp
# Commit: d33708e4f8ec
# Full Hash: d33708e4f8ecd98a449df50766a2ea59045b317f
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-29 09:50:10
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 3: Remove the unused MarkStackIter class r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160526
# ==============================================================================

diff -r 967fd2ce6f67 -r d33708e4f8ec js/src/gc/Marking.cpp
--- a/js/src/gc/Marking.cpp	Fri Oct 28 15:17:42 2022 +0000
+++ b/js/src/gc/Marking.cpp	Fri Oct 28 15:17:43 2022 +0000
@@ -1491,9 +1491,11 @@
   static const auto value = MarkStack::ScriptTag;
 };
 
+#ifdef DEBUG
 static inline bool TagIsRangeTag(MarkStack::Tag tag) {
   return tag == MarkStack::SlotsOrElementsRangeTag;
 }
+#endif
 
 inline MarkStack::TaggedPtr::TaggedPtr(Tag tag, Cell* ptr)
     : bits(tag | uintptr_t(ptr)) {
@@ -1716,51 +1718,6 @@
   return stack().sizeOfExcludingThis(mallocSizeOf);
 }
 
-MarkStackIter::MarkStackIter(MarkStack& stack)
-    : stack_(stack), pos_(stack.position()) {
-#ifdef DEBUG
-  stack.iteratorCount_++;
-#endif
-}
-
-MarkStackIter::~MarkStackIter() {
-#ifdef DEBUG
-  MOZ_ASSERT(stack_.iteratorCount_);
-  stack_.iteratorCount_--;
-#endif
-}
-
-inline size_t MarkStackIter::position() const { return pos_; }
-
-inline bool MarkStackIter::done() const { return position() == 0; }
-
-inline MarkStack::TaggedPtr MarkStackIter::peekPtr() const {
-  MOZ_ASSERT(!done());
-  return stack_.stack()[pos_ - 1];
-}
-
-inline MarkStack::Tag MarkStackIter::peekTag() const { return peekPtr().tag(); }
-
-inline void MarkStackIter::nextPtr() {
-  MOZ_ASSERT(!done());
-  MOZ_ASSERT(!TagIsRangeTag(peekTag()));
-  pos_--;
-}
-
-inline void MarkStackIter::next() {
-  if (TagIsRangeTag(peekTag())) {
-    nextArray();
-  } else {
-    nextPtr();
-  }
-}
-
-inline void MarkStackIter::nextArray() {
-  MOZ_ASSERT(TagIsRangeTag(peekTag()));
-  MOZ_ASSERT(position() >= ValueRangeWords);
-  pos_ -= ValueRangeWords;
-}
-
 /*** GCMarker ***************************************************************/
 
 /*
